### DISCUSSION TITLE
Does line operator work with JSON?

### DISCUSSION THREAD

**Discussion by treatmesubj** [mikefarah/yq#2018] [Q&A]
`yq` can find a key's value in my JSON file just fine per `yq '.sources["worker-nodes"][10].name' nodes.json`, but utilizing the [line operator](https://mikefarah.gitbook.io/yq/operators/line) via `yq '.sources["worker-nodes"][10].name | line' nodes.json` always returns `0`, rather than the line number. 
Does the `line` operator work with JSON files?

**Comment by mikefarah** [comment]
Sorry it only works with yaml files at the moment - the json parser doesn't give me that information :(

**Reply by treatmesubj** [reply]
I'm surprised I can't find any solutions or tools for it on the internet.
For text editors, I'd think people would want to jump to a key in a JSON file, which are often huge.
Like for working with YAML in Vim, I have a rudimentary function that utilizes `yq` to help me jump to the line of a key in a big YAML file, which is nice.
```vim
function YAMLGoToKey(key)
  let yqcmd = "yq '" . a:key . " | line' " . expand('%:p')
  let line = trim(system(yqcmd))
  execute "norm " . line . "gg"
endfunction
```

- [go-yaml issue](https://github.com/go-yaml/yaml/issues/108) -  adding line numbers to API
- no go-json [issues mentioning line numbers](https://github.com/goccy/go-json/issues?q=line+number)

specs and parsers for YAML & JSON are way over my head though, oh well.
